core/lib/zksync_core/src/lib.rs

```diff
@@ -828,8 +828,11 @@ async fn add_state_keeper_to_task_futures(
     batch_fee_input_provider: Arc<dyn BatchFeeModelInputProvider>,
     stop_receiver: watch::Receiver<bool>,
 ) -> anyhow::Result<()> {
-    let pool_builder = ConnectionPool::<Core>::singleton(postgres_config.master_url()?);
-    let state_keeper_pool = pool_builder
+    // One (potentially held long-term) connection for async catch up task and another connection for
+    // batch executor.
```

Comment:
Are 2 connections enough? Besides the batch executor (which AFAIU holds a connection for an entire L1 batch), and the catch up task (which holds a connection while it's active), we have `MempoolIO` that also needs to acquire connections from time to time while the batch is executing. I think it could be safer to:

- Construct a separate two-connection pool specifically for `AsyncRocksdbCache` (for it, we know that 2 connections is enough)
- Refactor `create_state_keeper()` to accept `AsyncRocksdbCache` (or `Arc<dyn ReadStorageFactory>`) as an arg. Still provide a singleton pool for `MempoolIO`.

Note that some other state keeper dependencies (e.g., `OutputHandler`) are already provided as args to `create_state_keeper()`, and both miniblock sealer and mempool fetcher use separate singleton pools.

Reply:
Done. My original thinking was that a singleton pool used to be enough for both batch executor and MempoolIO, but better safe than sorry indeed